In [30]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [31]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv.csv')
gdp_df.head(6)

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2019  2065.036398              NaN
1     Afghanistan  2018  2033.779002              NaN
2     Afghanistan  2017  2058.383832              NaN
3     Afghanistan  2016  2057.062164              NaN
4     Afghanistan  2015  2068.265904              NaN
5     Afghanistan  2014  2102.385234              NaN

In [32]:
internet_df = pd.read_csv('../data/internet_use.csv.csv', error_bad_lines = False)

b'Skipping line 4675: expected 4 fields, saw 6\n'


In [33]:
internet_df.head(6)

Country or Area  Year     Value  Value Footnotes
0     Afghanistan  2014  6.390000              NaN
1     Afghanistan  2013  5.900000              NaN
2     Afghanistan  2012  5.454545              NaN
3     Afghanistan  2011  5.000000              NaN
4     Afghanistan  2010  4.000000              NaN
5     Afghanistan  2009  3.550000              NaN

In [34]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6870 entries, 0 to 6869
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  6870 non-null   object 
 1   Year             6870 non-null   object 
 2   Value            6868 non-null   float64
 3   Value Footnotes  1 non-null      float64
dtypes: float64(2), object(2)
memory usage: 214.8+ KB


In [35]:
internet_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4675 entries, 0 to 4674
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  4675 non-null   object 
 1   Year             4675 non-null   object 
 2   Value            4495 non-null   float64
 3   Value Footnotes  948 non-null    float64
dtypes: float64(2), object(2)
memory usage: 146.2+ KB


#7 Look at the shape of each DataFrame - how many rows? How many columns?

gdp_df has 6870 rows and 4 columns
internet_df has 4675 rows and 4 columns

#8 Take a look at the datatypes for the columns in each DataFrame.

In [36]:
gdp_df.dtypes

Country or Area     object
Year                object
Value              float64
Value Footnotes    float64
dtype: object

In [37]:
internet_df.dtypes

Country or Area     object
Year                object
Value              float64
Value Footnotes    float64
dtype: object

#9 Take a look at the last 10 rows of each DataFrame in turn.

In [38]:
gdp_df.tail(10)

Country or Area                                      Year        Value  \
6860        Zimbabwe                                      1997  3580.048793   
6861        Zimbabwe                                      1996  3520.430146   
6862        Zimbabwe                                      1995  3226.413930   
6863        Zimbabwe                                      1994  3263.934978   
6864        Zimbabwe                                      1993  3033.504852   
6865        Zimbabwe                                      1992  3054.889178   
6866        Zimbabwe                                      1991  3426.598094   
6867        Zimbabwe                                      1990  3324.348171   
6868   footnoteSeqID                                  Footnote          NaN   
6869               2  Excludes South Sudan after July 9, 2011.          NaN   

      Value Footnotes  
6860              NaN  
6861              NaN  
6862              NaN  
6863              NaN  
6864              NaN  
6865              NaN  
6866              NaN  
6867              NaN  
6868              NaN  
6869              NaN

In [39]:
internet_df.tail(10)

Country or Area                                               Year  \
4665             170                   Refers to the total population.    
4666             171                        Internet Dial-up customers.   
4667             172  Population age 16+ using the Internet in the l...   
4668             173  Population age 16+ using internet in the last ...   
4669             174  U.S. Census Bureau, Table 2. Reported Internet...   
4670             175             Includes individuals 3 years and older   
4671             176                                   NTIA/CPS survey.   
4672             178  Estimated based on Survey's results. Populatio...   
4673             179                     Preliminary. Country estimate.   
4674             180  The methodology used to estimated the figure f...   

      Value  Value Footnotes  
4665    NaN              NaN  
4666    NaN              NaN  
4667    NaN              NaN  
4668    NaN              NaN  
4669    NaN              NaN  
4670    NaN              NaN  
4671    NaN              NaN  
4672    NaN              NaN  
4673    NaN              NaN  
4674    NaN              NaN

#10 Drop the value footnotes column from both DataFrames. Check that this worked as expected.

In [40]:
del gdp_df['Value Footnotes']
gdp_df.head()

Country or Area  Year        Value
0     Afghanistan  2019  2065.036398
1     Afghanistan  2018  2033.779002
2     Afghanistan  2017  2058.383832
3     Afghanistan  2016  2057.062164
4     Afghanistan  2015  2068.265904

In [41]:
del internet_df['Value Footnotes']
internet_df.head()

Country or Area  Year     Value
0     Afghanistan  2014  6.390000
1     Afghanistan  2013  5.900000
2     Afghanistan  2012  5.454545
3     Afghanistan  2011  5.000000
4     Afghanistan  2010  4.000000

#11 Change the columns for the GDP Per Capita DataFrame to ‘Country’, ‘Year’, and ‘GDP_Per_Capita’.